libs/pika/execution/include/pika/execution/algorithms/drop_operation_state.hpp

```diff
@@ -248,5 +249,16 @@ namespace pika::execution::experimental {
         {
             return detail::partial_algorithm<drop_operation_state_t>{};
         }
-    } drop_operation_state{};
+    };
+
+    /// \brief Releases the operation state of the adaptor before signaling a connected receiver.
+    /// The operation_state of previous senders can potentially hold on to a allocated memory longer
+    /// than necessary which would prevent other algorithm from accessing this resource.
+    ///
+    /// Sender adaptor that takes any sender and returns a sender. Values sent as references will be
+    /// copied.
```

Comment:
I would maybe still keep this as the first thing in the detailed documentation (i.e. after the `\brief`) and then move the motivation for it at the end?